### ISSUE TITLE
todos

### ISSUE THREAD

**Issue by jasmainak** [autoreject/autoreject#5]
[] remove joblibs dependency
[] add more narrative to documentations
[] add tests

**Comment by jona-sassenhagen** [comment]
Verbose param would be nice indeed! It's very talkative.

**Comment by jasmainak** [comment]
indeed! I hope to get around to it once @dengemann reads the paper draft ;)

Of course, feel free to make a PR if you have a time. How does it work on your data?

**Comment by jona-sassenhagen** [comment]
I only started using it today. I will report when I actually have some results.

**Comment by jona-sassenhagen** [comment]
A preliminary result is that this really helps with getting good ICA decompositions, and I have seen slightly improved decoding accuracies on one data set.

Verbosity param is super important though!

And, less important, save and load methods.

**Comment by jasmainak** [comment]
okay, thanks for the feedback!

I have in mind using something like `tqdm` to reduce verbosity. Would that work for you? I am a bit hesitant about using logger just yet because I had trouble setting it up last I tried.

By save and load methods, you mean you want to be able to save the rejected trials / thresholds? That's an interesting thought indeed!

**Comment by jona-sassenhagen** [comment]
>I have in mind using something like tqdm to reduce verbosity. Would that work for you? I am a bit hesitant about using logger just yet because I had trouble setting it up last I tried.

Well my problem is, I'm running this on 30 subjects and I get hundreds of lines of logging. It really clutters up my notebooks, so much that it actually slows down the system. I think in the end you'll need a verbosity flag.

>By save and load methods, you mean you want to be able to save the rejected trials / thresholds? That's an interesting thought indeed!

The fitted autorej objects. Currently I pickle them.

**Comment by jona-sassenhagen** [comment]
Ok, after having used it for 3 weeks straight:
1. it's pretty good, I use it for everything now
2. rrrrrrreally needs a verbosity flag. It's basically unusable in the notebook without silencing otherwise.
3. A write method would still be appreciated.

**Comment by jasmainak** [comment]
okay, thanks for the feedback! I am at the moment experiencing [thrashing](http://stackoverflow.com/a/20842853/4018756). Too many things on the plate. So, I can't promise about the `write` method soon but I hope to solve your urgent need with `verbosity` :) I'll add the rest to my todo list ...

**Comment by nbara** [comment]
Would you consider implementing some way of saving the AR cross validation results to disk? (or is that already possible?)

`ar = LocalAutoRejectCV(..)
ar.save()`

**Comment by jona-sassenhagen** [comment]
For now you can pickle. But save and loadMethods would be nice.

**Comment by jasmainak** [comment]
Thanks for the feedback @nbara. It's next on my todo list for autoreject :) I need to figure out what's the best way to do it before jumping into it.

**Comment by jona-sassenhagen** [comment]
save would still be nice :)
